api-test/hasura/actions/_handlers/updateCircleStartingGive.test.ts

```diff
@@ -0,0 +1,132 @@
+import { DateTime } from 'luxon';
+
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import {
+  createCircle,
+  createProfile,
+  createUser,
+  mockUserClient,
+} from '../../../helpers';
+import { getUniqueAddress } from '../../../helpers/getUniqueAddress';
+
+const now = DateTime.now();
+const DURATION_IN_DAYS = 3;
+let address1, address2, profile, client, circle;
+
+beforeEach(async () => {
+  address1 = await getUniqueAddress();
+  address2 = await getUniqueAddress();
+  circle = await createCircle(adminClient);
+  profile = await createProfile(adminClient, { address: address1 });
```

Comment:
you can just create a profile, and then get the auto-generated address from it, rather than creating `address1` first and then passing it through.